### PR TITLE
[proxy] Don't forward empty `options` to compute nodes

### PR DESCRIPTION
**Pull request by funbringer**
Clients may specify endpoint/project name via `options=project=...`, so we should not only remove `project=` from `options` but also drop `options` entirely, because connection pools don't support it.

Discussion: https://neondb.slack.com/archives/C033A2WE6BZ/p1676464382670119

### CODE REVIEW

**Comment by Daniel-ef** [comment]
Could you remove me from the CODE_OWNERS? :) I'm not sure it makes sense to keep me there.

File: proxy/src/compute.rs [hunk 1]
```diff
@@ -225,3 +220,46 @@ impl ConnCfg {
...
+
+        // It's likely that clients will only use options to specify endpoint/project.
+        let params = StartupMessageParams::new([("options", "project=foo")]);
+        assert_eq!(filtered_options(&params), None);
+
+        // Same, because unescaped whitespaces are no-op.
+        let params = StartupMessageParams::new([("options", " project=foo ")]);
+        assert_eq!(filtered_options(&params).as_deref(), None);
+
+        let params = StartupMessageParams::new([("options", r"\  project=foo \ ")]);
```
Comment (ololobus) [inline proxy/src/compute.rs:259]: I wonder is it libpq / client lib responsible for converting from url-safe encoding like `?options=project%3Dep-raspy-unit-343115%20something` to this `\` escaping?

**Review by ololobus — APPROVED** [review]
New logic looks good to me

**Comment by ololobus** [comment]
> Could you remove me from the CODE_OWNERS? :) I'm not sure it makes sense to keep me there.

I think it's rather question to @vadim2404. Current GH teams are a bit messy, indeed

**Comment by ololobus** [comment]
But I think it's generally good that people from console / control-plane are aware of new proxy changes. It doesn't necessary mean that you need to thoroughly review Rust code (I cannot do that for sure)

File: proxy/src/compute.rs (hunk 1 above)
Comment (funbringer) [reply proxy/src/compute.rs:259]: Correct!

Examples:
* https://github.com/postgres/postgres/blob/5352ca22e0012d48055453ca9992a9515d811291/src/interfaces/libpq/fe-connect.c#L5961
* https://github.com/lib/pq/blob/922c00e176fb3960d912dc2c7f67ea2cf18d27b0/url.go#L32
* https://github.com/sfackler/rust-postgres/blob/4bae134f501db5c08b8dd43a44474e515bbb39d3/tokio-postgres/src/config.rs#L921

**Pull request merged**